include/effects/matrix/spectrumeffects.h

```diff
@@ -329,6 +349,36 @@ class SpectrumAnalyzerEffect : public LEDStripEffect, virtual public VUMeterEffe
     {
     }
 
+    SpectrumAnalyzerEffect(const JsonObjectConst& jsonObject)
+        : LEDStripEffect(jsonObject), 
+          _numBars(jsonObject["nmb"]),
+          _colorOffset(0),
+          _scrollSpeed(jsonObject[PTY_SPEED]), 
+          _fadeRate(jsonObject["frt"]),
+          _palette(jsonObject[PTY_PALETTE].as<CRGBPalette16>()),
+          _peak1DecayRate(jsonObject["pd1"]),
+          _peak2DecayRate(jsonObject["pd2"])

```

Comment:
Cool, I like how it can use the incoming object to init all of its params...

Reply:
I do too. :)